include/deal.II/matrix_free/operators.h

```diff
@@ -328,6 +328,101 @@ namespace MatrixFreeOperators
 
 
 
+  /**
+   * This class implements the operation of the action of a Laplace matrix.
+   *
+   * @author Denis Davydov, 2016
+   */
+  template <int dim, int fe_degree, int n_q_points_1d = fe_degree+1, int n_components = 1, typename Number = double>
+  class LaplaceOperator : public Base<dim, Number>
+  {
+  public:
+
+    /**
+     * Constructor.
+     */
+    LaplaceOperator ();
+
+    /**
+     * The diagonal is approximated by computing a local diagonal matrix per element
+     * and distributing it to the global diagonal. This will lead to wrong results
+     * on element with hanging nodes but is still an acceptable approximation
+     * to be used in preconditioners.
+     */
+    virtual void compute_diagonal ();
+
+    /**
+     * Set heterogeneous scalar coefficient @p scalar_coefficient to be used at
+     * the quadrature points. The Table should be of correct size, consistent
+     * with the total number of quadrature points in <code>dim</code>-dimensions,
+     * controlled by @p n_q_points_1d template parameter. Here,
+     * <code>(*scalar_coefficient)(cell,q)</code> corresponds to the value of the
+     * coefficient, where <code>cell</code> is the cell number and <code>q</code>
+     * is the number of quadrature point.
```

Comment:
that's exactly what it is here, `cell` is NOT active cell index, but a cell (or butch cell) number. AFAIK, there is no easy way of going from cell iterator and cell number.